internal/compiler/typeloader.rs

```diff
@@ -302,16 +293,69 @@ impl TypeLoader {
         import_token: Option<NodeOrToken>,
         mut import_stack: HashSet<PathBuf>,
     ) -> Option<PathBuf> {
-        let (path_canon, builtin) =
-            { state.borrow().tl.resolve_import_path(import_token.as_ref(), file_to_import) };
+        let mut borrowed_state = state.borrow_mut();
+
+        let (path_canon, builtin) = match borrowed_state
+            .tl
+            .resolve_import_path(import_token.as_ref(), file_to_import)
+        {
+            Some(x) => x,
+            None => match dunce::canonicalize(file_to_import) {
+                Ok(path) => {
+                    if import_token.as_ref().and_then(|x| x.source_file()).is_some() {
+                        borrowed_state.diag.push_warning(
+                        format!(
+                            "Loading \"{file_to_import}\" relative to the work directory is deprecated. Files should be imported relative to their import location",
+                        ),
+                        &import_token,
+                    );
+                    }
+                    (path, None)
+                }
+                Err(err) if err.kind() == std::io::ErrorKind::NotFound => {
+                    // We will load using the `open_import_fallback`
+                    // Simplify the path to remove the ".."
+                    let mut path = import_token
+                        .as_ref()
+                        .and_then(|tok| tok.source_file().map(|s| s.path()))
+                        .and_then(|p| p.parent())
+                        .map_or(PathBuf::new(), |p| p.into());
+                    for c in Path::new(file_to_import).components() {
+                        use std::path::Component::*;
+                        match c {
+                            RootDir => path = PathBuf::new(),
+                            CurDir => {}
+                            ParentDir
+                                if matches!(
+                                    path.components().last(),
+                                    Some(Normal(_) | RootDir,)
```

Comment:
This looks like syntax sorcery to me :-)

`matches!(blah, Some(Normal(_) | RootDir,))`

Is the last comma part of the pattern syntax and ignored?

Reply:
I didn't do that on purpose. 
But this code was reformated several times as I edited. But I'll fix